src/components/utils/file.js

```diff
@@ -1,16 +1,50 @@
 import { findKey } from '~utils/object'
-import { isOfMimeType } from '~utils/blob'
+import { isOfMimeType, canvasToBlob } from '~utils/blob'
 
-const defaultFileTypesAccepted = ['jpg', 'jpeg', 'png', 'pdf']
-const maxFileSizeAcceptedByAPI = 10000000 // The Onfido API only accepts files below 10 MB
+const DEFAULT_ACCEPTED_FILE_TYPES = ['jpg', 'jpeg', 'png', 'pdf']
+const MAX_FILE_SIZE_ACCEPTED_BY_API = 10000000 // The Onfido API only accepts files below 10 MB
 export const validateFileTypeAndSize = (
   file,
-  acceptedTypes = defaultFileTypesAccepted
+  acceptedTypes = DEFAULT_ACCEPTED_FILE_TYPES
 ) =>
   findKey(
     {
       INVALID_TYPE: (file) => !isOfMimeType(acceptedTypes, file),
-      INVALID_SIZE: (file) => file.size > maxFileSizeAcceptedByAPI,
+      INVALID_SIZE: (file) => file.size > MAX_FILE_SIZE_ACCEPTED_BY_API,
     },
     (checkFn) => checkFn(file)
   )
+
+export const resizeImageFile = (file, onImageResize) => {
+  const reader = new FileReader()
+  reader.onload = (readerEvent) => {
+    const image = new Image()
+    image.onload = () => {
+      // We want to resize to 720p (1280×720px)
+      const MAX_SIZE_IN_PIXEL = 1280
+      let width = image.width
+      let height = image.height
+      const WIDTH_ONE_PERCENT = width / 100
+      const HEIGHT_ONE_PERCENT = height / 100
+      let imageCurrentPercent
+      if (width > height) {
+        // landscape orientation
+        width = MAX_SIZE_IN_PIXEL
+        imageCurrentPercent = width / WIDTH_ONE_PERCENT
+        height = HEIGHT_ONE_PERCENT * imageCurrentPercent
+      } else {
+        // portrait orientation
+        height = MAX_SIZE_IN_PIXEL
+        imageCurrentPercent = height / HEIGHT_ONE_PERCENT
+        width = WIDTH_ONE_PERCENT * imageCurrentPercent
+      }
+      const tempCanvas = document.createElement('canvas')
+      tempCanvas.width = width
+      tempCanvas.height = height
+      tempCanvas.getContext('2d').drawImage(image, 0, 0, width, height)
```

Comment:
If I understand this code correctly you crop the image here instead of resizing, is that intentional?

Reply:
No, we don't want it to be cropped. With this the image is being scaled to the resized dimensions. The image would indeed be cropped if I had specified it as follows:
```
tempCanvas
        .getContext('2d')
        .drawImage(image, 0, 0, width, height, 0, 0, width, height)
```
I probably should name the resized height & width better.